libcobblersignatures/models/osversion.py

```diff
@@ -67,124 +67,134 @@ def __eq__(self, other):
+    def signatures(self) -> list:
         return self._signatures
 
     @signatures.setter
-    def signatures(self, value):
-        self._signatures = value
+    def signatures(self, value: list):
+        if isinstance(value, list):
+            self._signatures = value
 
     @signatures.deleter
     def signatures(self):
-        del self._signatures
+        self._signatures = []
 
     @property
-    def version_file(self):
+    def version_file(self) -> str:
         return self._version_file
 
     @version_file.setter
     def version_file(self, value):
-        self._version_file = value
+        if isinstance(value, str):
+            self._version_file = value
 
     @version_file.deleter
     def version_file(self):
-        del self._version_file
+        self._version_file = ""
 
     @property
-    def version_file_regex(self):
+    def version_file_regex(self) -> str:
         return self._version_file_regex
 
     @version_file_regex.setter
     def version_file_regex(self, value):
-        self._version_file_regex = value
+        if isinstance(value, str):
+            self._version_file_regex = value
 
     @version_file_regex.deleter
     def version_file_regex(self):
-        del self._version_file_regex
+        self._version_file_regex = ""
 
     @property
-    def kernel_arch(self):
+    def kernel_arch(self) -> str:
         return self._kernel_arch
 
     @kernel_arch.setter
     def kernel_arch(self, value):
-        self._kernel_arch = value
+        if isinstance(value, str):
+            self._kernel_arch = value
 
     @kernel_arch.deleter
     def kernel_arch(self):
-        del self._kernel_arch
+        self._kernel_arch = ""
 
     @property
-    def kernel_arch_regex(self):
+    def kernel_arch_regex(self) -> str:
         return self._kernel_arch_regex
 
     @kernel_arch_regex.setter
     def kernel_arch_regex(self, value):
-        self._kernel_arch_regex = value
+        if isinstance(value, str):
+            self._kernel_arch_regex = value
 
     @kernel_arch_regex.deleter
     def kernel_arch_regex(self):
-        del self._kernel_arch_regex
+        self._kernel_arch_regex = ""
 
     @property
-    def supported_arches(self):
+    def supported_arches(self) -> list:
         return self._supported_arches
 
     @supported_arches.setter
     def supported_arches(self, value):
-        self._supported_arches = value
+        if isinstance(value, list):
+            self._supported_arches = value
 
     @supported_arches.deleter
     def supported_arches(self):
-        del self._supported_arches
+        self._supported_arches = []
 
     @property
-    def supported_repo_breeds(self):
+    def supported_repo_breeds(self) -> list:
         return self._supported_repo_breeds
 
     @supported_repo_breeds.setter
     def supported_repo_breeds(self, value):
-        self._supported_repo_breeds = value
+        if isinstance(value, list):
+            self._supported_repo_breeds = value
```

Comment:
This is silently not setting the value if it's not a list.

Reply:
Now we raise a TypeError()